lms/static/scripts/frontend_apps/components/SubmitGradeForm.js

```diff
@@ -49,13 +50,17 @@ const useFetchGrade = student => {
       const fetchData = async () => {
         setGradeLoading(true);
         setGrade(''); // Clear previous grade so we don't show the wrong grade with the new student
-
-        const response = await fetchGrade({ student, authToken });
-        if (!didCancel) {
-          // Only set these values if we didn't cancel this request
-          if (response.currentScore) {
-            setGrade(scaleGrade(response.currentScore, GRADE_MULTIPLIER));
+        try {
+          const response = await fetchGrade({ student, authToken });
+          if (!didCancel) {
+            // Only set these values if we didn't cancel this request
+            if (response.currentScore) {
+              setGrade(scaleGrade(response.currentScore, GRADE_MULTIPLIER));
+            }
           }
```

Comment:
I haven't tested this, but I wonder if this is simpler:

```suggestion

       if (didCancel) { return; }
       
        try {
          const response = await fetchGrade({ student, authToken });
          if (response.currentScore) {
              setGrade(scaleGrade(response.currentScore, GRADE_MULTIPLIER));
          }
```

Reply:
I think thats reasonable!